ksql-engine/src/test/java/io/confluent/ksql/engine/KsqlEngineTest.java

```diff
@@ -383,82 +383,6 @@ public void shouldHandleCommandsSpreadOverMultipleLines() {
     assertThat(parsedStatements, hasSize(1));
   }
 
-  @Test
-  public void shouldCleanupSchemaAndTopicForStream() throws Exception {
```

Comment:
Would be good to see test deleted tests added as functional tests in `KsqlResourceTest` to just ensure everything is wired together properly. i.e. add tests that:
 - test `DROP TABLE x  DELETE TOPIC;` for non-avro deletes the topic and posts the correct thing to the command topic.
 - test `DROP TABLE x  DELETE TOPIC;` for avro deletes the topic & schema and posts the correct thing to the command topic.
 - test `DROP TABLE` doesn't delete topic or schema
 - test negative cases, e.g. table does not exist, topic does not exist, schema does not exist, etc.
 - test all above for streams too.

Reply:
There were added into `TopicDeleteInjectorTest` - I believe having a test in `KsqlResourcetest` that makes sure that `TopicDeleteInjector` is called is sufficient. Otherwise we have the extra burden of duplicated tests.

I will add the negative cases into `TopicDeleteInjectorTest`.

Reply:
Having the test that I mentioned is a little tricky... I will do that alongside the follow-up mentioned above with RecoveryTest to ensure that the correct injectors are called

Reply:
So `TopicDeleteInjectorTest` is a _unit_ test, and has good coverage.  What I'm suggesting we add is a high level _functional_ test, (which is kind of what's been removed here), to prove everything has been wired up.

IMHO and experience, only having unit tests that prove units of functionality work and are wired up isn't generally sufficient. It's still possible for the functionality to not work.  Hence, suggesting we need functional tests for the main scenarios.

Reply:
Added coverage into `RestApiTest`